Library/Homebrew/cli_parser.rb

```diff
@@ -162,6 +162,8 @@ def formula_options
             end
           end
         end
+      rescue FormulaUnavailableError
+        []
```

Comment:
Just need to ignore this exception here. Found no alternative.

Reply:
This can lead to some tricky buggy cases. 
What if the user passes multiple formulae names with a single invalid name ?
`ARGV.formulae` in that case would throw an exception and none of the formula options would get declared. 
@MikeMcQuaid  I think we could modify `ARGV.formulae` 's code to better handle this.

Reply:
@GauthamGoli Yes, that seems like a good idea. As we discussed I think almost all of https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/ARGV.rb would move to https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cli_parser.rb instead.